pkg/util/cloudproviders/cloudfoundry/cloudfoundry_test.go

```diff
@@ -74,6 +74,7 @@ func TestHostAliasDefault(t *testing.T) {
 	mockHostname := "hostname"
 
 	// mock getFqdn to avoid flakes in CI runners
+	defer func() { getFqdn = netutil.Fqdn }()
```

Comment:
The mock here should be reset to the global variable's original implementation:
https://github.com/DataDog/datadog-agent/blob/20d2e14eb6201956bacbb6eb4c010df7a95be1e1/pkg/util/cloudproviders/cloudfoundry/cloudfoundry.go#L25